src/services/tx/safeUpdateParams.ts

```diff
@@ -1,11 +1,41 @@
 import { MetaTransactionData, OperationType } from '@gnosis.pm/safe-core-sdk-types'
 import { ChainInfo, SafeInfo } from '@gnosis.pm/safe-react-gateway-sdk'
-import { getFallbackHandlerContractInstance, getGnosisSafeContractInstance } from '@/services/contracts/safeContracts'
+import {
+  getFallbackHandlerContractInstance,
+  getGnosisSafeContractInstance,
+  _getSafeContractDeployment,
+} from '@/services/contracts/safeContracts'
 import { LATEST_SAFE_VERSION } from '@/config/constants'
+import { isValidSafeVersion } from '@/hooks/coreSDK/safeCoreSDK'
+import { Gnosis_safe__factory } from '@/types/contracts/factories/Gnosis_safe__factory'
+import { getWeb3 } from '@/hooks/wallets/web3'
 
 export const CHANGE_MASTER_COPY_ABI = 'function changeMasterCopy(address _masterCopy)'
 export const CHANGE_FALLBACK_HANDLER_ABI = 'function setFallbackHandler(address handler)'
 
+const getContractInterface = (chain: ChainInfo, version: string) => {
+  if (isValidSafeVersion(version)) {
+    return getGnosisSafeContractInstance(chain, version).contract.interface
+  }
+
+  // < 1.1.1 Safes are not supported by Core SDK. We must connect to the contract directly
+
+  const safeDeployment = _getSafeContractDeployment(chain, version)
+  const contractAddress = safeDeployment?.networkAddresses[chain.chainId]
+
+  if (!contractAddress) {
+    throw new Error(`GnosisSafe contract not found for chainId: ${chain.chainId}`)
+  }
+
+  const web3 = getWeb3()
+
+  if (!web3) {
+    throw new Error('No provider instance found')
+  }
+
+  return Gnosis_safe__factory.connect(contractAddress, web3).interface
```

Comment:
If we only want to encode the function call we can use the interface directly without relying on `web3` i.e. `return Gnosis_safe__factory.createInterface()`